apps/imageHotspotCreator/src/components/ImageSelector/selectImage.tsx

```diff
@@ -24,14 +23,35 @@ const SelectImage = ({
 }: any) => {
   //state Declaratios
   const [imageFile, setImageFile] = useState<any>();
-  const [uploadYourImage, setUploadYourImage] = useState<any>(true);
+  const [uploadYourImage, setUploadYourImage] = useState(true);
   const [uploadAsLink, setUploadAsLink] = useState("");
   const [uploadExisting, setUploadExisting] = useState("");
 
   const [imageInExisting, setImageInExisting] = useState<any>(null);
 
   const [showUrlPreview, setShowUrlPreview] = useState(false);
   const [urlInput, setUrlInput] = useState("");
+  const [missedField, setMissedField] = useState([]);
+  const [defaultLocale, setDefaultLocale] = useState("en-US");
+
+  useEffect(() => {
+
+    const requiredNames = ["Title", "Image URL", "Hotspots"];
+    const missedObject: any = cloneDeep(missedField);
+
+    // Check if required names exist and create objects accordingly
+    for (const name of requiredNames) {
+      const found = sdk.contentType.fields.find((obj: any) => obj.name === name); // Check if object with name exists
+      if (!found) {
+        missedObject.push({ name }); // Add missing object with just the name property
+      }
+    }
+    if (missedObject.length > 0) {
+      setMissedField(missedObject);
+    }
+
+    setDefaultLocale(sdk?.locales?.default);
+  }, [])
```

Comment:
Same question about the dependency array: is it intentionally empty?

Reply:
Yes, intentionally.